testdata/project-v3/main.go

```diff
@@ -49,12 +49,12 @@ func init() {
 
```

Comment:
IMO we need to create a new testdata_generate with this scenario as well such as we have for multigroup and addon. See: https://github.com/kubernetes-sigs/kubebuilder/blob/master/generate_testdata.sh

WDYT?

Reply:
Because this param is on the `init` should I have testdata generated for addon + config and multi-group + config and config without any other options? or should I just create a single case with config and no other options?

Reply:
IMHO you should only have 1 testdata set for this, not 3. Multigroup and addon don't affect the component-config setup, and adding more testdata folders add way more load to the test done in CI for very push. Bear in mind that for every testdata folder, we are creating it, checking that every file created and the ones in the repo are exactly the same character by character, and then performing their tests for several platforms (Linux and MacOs in GH Actions, and I think several others in Prow).